test/integration/compute/test_addresses.rb

```diff
@@ -0,0 +1,84 @@
+require "helpers/integration_test_helper"
+require "helpers/client_helper"
+require "securerandom"
+
+class TestComputeAddresses < FogIntegrationTest
+  DEFAULT_REGION = "us-central1".freeze
+  ADDRESS_RESOURCE_PREFIX = "fog-test-address".freeze
+
+  include ClientHelper
+
+  # Ensure we clean up any created resources
+  Minitest.after_run do
+    client = Fog::Compute::Google.new
+    addresses = client.list_addresses(DEFAULT_REGION)[:body]["items"]
```

Comment:
Again, I wanted to keep things on the raw api call level here since this test is about exactly that; it tends to make it easier to read.